### PR TITLE
Lazily instantiate asyncio event loop; use currently running asyncio event loop if there is one

### PR DESCRIPTION
**Pull request by michelcrypt4d4mus**
As it is currently stands instantiating `DesktopNotifierSync()` results in an attempt to create a new `asyncio` event loop. This is not ideal if 
1. `desktop-notifier` library is being used in a larger project that is trying to do its own event loop management
2. `DesktopNotifierSync` is eagerly instantiated when the application is launched (my app was instantiating `DesktopNotifierSync() at launch time as more or less a singleton)

## Changes

This PR contains a small change that:
1. **makes `DesktopNotifierSync`'s event loop instantiation lazy:** it only bothers to find or construct an event loop once `send()` or similar is called by the application
2. **attempts to use the currently executing event loop if there is one:** there are some issues with this though, see below.

Theoretically this change would allow someone to use `DesktopNotifierSync` in a context where some other piece of code has already launched an `asyncio` event loop. Of course that might be of limited utility (if the app is already running an event loop why would you use `DesktopNotifierSync` as opposed to just doing it all asynchronously?) but given that i can see a situation where someone using the lib might want to do it (for instance let's say the event loop management is done by some other library they are relying on, which is more or less my use case) but it is admittedly pretty marginal.

## Issues

Unfortunately in my particular context of a [Twisted](https://github.com/twisted/twisted) networking app the code in this PR doesn't actually work. It just hangs indefinitely waiting for `future.result()`. i'm pretty sure (though not 100% sure) that this is the fault of Twisted which is a 20 year old networking library that implemented asynchronicity in python literally decades before `asyncio` even existed and as a result [hates it when you try to do any thread management outside of its purview](https://docs.twisted.org/en/twisted-18.7.0/core/howto/threading.html) but i could just be making a mistake... either way I figured it might be worth opening this PR just because lazy instantiation of `DesktopNotifierSync._loop` could prevent an event loop collision should someone attempt to use `DesktopNotifierSync` in a situation _with_ a running event loop but _without_ a `twisted.reactor` trying to manage threading and everything else.

To get `DesktopNotifierSync.send()` to work in my particular context I had to switch to using `asyncio.create_task()` instead of `asyncio.run_coroutine_threadsafe()` and then just discard the result of the coroutine via `return None` ([branch comparison view for these changes](https://github.com/michelcrypt4d4mus/desktop-notifier/compare/asyncio_delay_loop_start...michelcrypt4d4mus:desktop-notifier:asyncio_delay_loop_start_use_running_loop?expand=1)).  That's obviously less than ideal but at least it proves that it can be done.

## UPDATE: Example
This short block of code shows how this works in a situation where the event loop is created outside of `desktop-notifier`:
```python
import asyncio
from desktop_notifier.sync import DesktopNotifierSync
from time import sleep


loop = asyncio.new_event_loop()
print("Created event loop, sleeping...")
sleep(1)
print("Triggering notification...")

notifier = DesktopNotifierSync()
notifier.send(title='Nasir Jones', message='The World is Yours')
print("Triggered notification, starting event loop!")
loop.run_forever()
```

though i haven't tested trying to use `DesktopNotifierSync` in a situation where the event loop has been _both_ created _and_ started via `loop.run_forever()` or similar.

### CODE REVIEW

**Comment by samschott** [comment]
I'm not entirely sure why you want to use the synchronous version of the API if you already have a running asyncio event loop.

Since you loose all callback functionality _and_ end up blocking other code, the async API seems like the better choice in cases where `asyncio.get_running_loop()` would return an event loop in the first place.

Though I am not at all familiar with Twisted, which seems to be the targeted use case. Why is it not possible to just use the async API with Twisted?

**Comment by michelcrypt4d4mus** [comment]
> _I'm not entirely sure why you want to use the synchronous version of the API if you already have a running asyncio event loop._

the funny thing is that i don't actually - except for one small task i have to run when bootstrapping my application i use the asynchronous version and in fact that's part of why i chose this library, but messing around with that small piece i ended up with this change which might be a small improvement for other people in other situations just bc of the lazy instantiation.

> _is it not possible to just use the async API with Twisted?_

it is and that's exactly what i'm doing (though it is a bit janky).

this PR doesn't really help me out and you should feel free to just close it, though at the same time i don't think there's any downside to checking for a running event loop in a lazy way instead of in an eager way.

**Comment by michelcrypt4d4mus** [comment]
i cleaned up the linter issue and removed the `[WIP]` from the title bc i guess this isn't really WIP.

tldr i don't need this PR for my use case but i do think it's slightly nicer to do a lazy instantiation of the event loop and allow for a running loop. feel free to just close if you don't want this change.

**Comment by samschott** [comment]
As discussed, I don't really see the value of this since clients with a running asyncio eventloop should use the async API instead.

If there are any issues that prevent them from doing so, I'd rather try to address those.
